cores/esp8266/uart.cpp

```diff
@@ -509,7 +509,10 @@ uart_tx_fifo_full(const int uart_nr)
 static void
 uart_do_write_char(const int uart_nr, char c)
 {
-    while(uart_tx_fifo_full(uart_nr));
+    while(uart_tx_fifo_full(uart_nr))
+    {
+        optimistic_yield(1000UL);
+    }
```

Comment:
Busywaiting on a fifo full has a mean expectation time of (10/2) bits @ baud.  For 115.2k that's 4.34us. I think this is a no-op in general.  Even at 9600bps it's only 520us, so same conclusion.  So this will generally be a noop...

Reply:
@earlephilhower Please look at this again from this perspective: `uart_do_write_char()` gets called from `uart_write_char()`, where before there is an `optimistic_yield(10000UL);` and `uart_write()`, where there is none, despite the latter being the longer running block write.
Might be good to do the same thing in both places. I'm going for 10000µs, perhaps it's better unconditionally after the register write in `uart_do_write_char()`, what do you think?

Reply:
OK, sounds reasonable.